docs/reproject.rst

```diff
@@ -0,0 +1,114 @@
+
+.. code-block:: python
+
+  >>> # Create a target WCS object with new transformations.
+  >>> target_wcs = astropy.wcs.WCS(naxis=3)
+  >>> target_wcs.wcs.ctype = 'WAVE', 'HPLT-TAN', 'HPLN-TAN'
+  >>> target_wcs.wcs.cunit = 'Angstrom', 'deg', 'deg'
+  >>> target_wcs.wcs.cdelt = 0.1, 0.5, 0.4
+  >>> target_wcs.wcs.crpix = 0, 2, 2
+  >>> target_wcs.wcs.crval = 10, 0.5, 1
+
+  >>> reprojected_cube = my_cube.reproject_to(target_wcs=target_wcs, shape_out=(4, 4, 10))
+
+In the above example, the ``CDELT1`` parameter of the ``target_wcs`` was modified to 0.1 (from 0.2). This allowed us to upsample the wavelength axis by a factor of 2.
+Accordingly, the shape of the data was updated to ``(4, 4, 10)`` from the initial shape ``(4, 4, 5)``. The wavelength axis goes last since the pixel shape and WCS shape are represented in opposite directions.
+
+Currently, this method does not handle the ``mask``, ``extra_coords``, and ``uncertainty`` attributes. These values are dropped from the returned `~ndcube.NDCube`.
+
+.. _cube_sequence_reproject:
+
+Reprojecting an NDCubeSequence
+==============================
+
+`~ndcube.NDCubeSequence` offers an interesting application of this functionality. The `~ndcube.NDCubeSequence.combine_cubes` method reprojects all the cubes in an `~ndcube.NDCubeSequence` to a common WCS object, and then combines them all to return a single `~ndcube.NDCube`.
+It does this by extracting the sequence axis of the `~ndcube.NDCubeSequence` and adding it to the WCS, hence creating an (N+1)-dimensional `~ndcube.NDCube`. This extra dimension can be used just like any other existing axes.
+
+This method can be used as:
+
+.. expanding-code-block:: python
+  :summary: Expand to see <span class="pre">my_cube</span> instantiated.
+
+  >>> import astropy.wcs
+  >>> import numpy as np
+  >>> from ndcube import NDCubeSequence
+
+  >>> # Define data arrays.
+  >>> data0 = np.random.rand(4, 4, 5)
+  >>> data1 = np.random.rand(4, 4, 5)
+  >>> data2 = np.random.rand(4, 4, 5)
+
+  >>> # Define WCS transformations in an astropy WCS object.
+  >>> wcs0 = astropy.wcs.WCS(naxis=3)
+  >>> wcs0.wcs.ctype = 'WAVE', 'HPLT-TAN', 'HPLN-TAN'
+  >>> wcs0.wcs.cunit = 'Angstrom', 'deg', 'deg'
+  >>> wcs0.wcs.cdelt = 0.2, 0.5, 0.4
+  >>> wcs0.wcs.crpix = 0, 2, 2
+  >>> wcs0.wcs.crval = 10, 0.5, 1
+
+  >>> wcs1 = astropy.wcs.WCS(naxis=3)
+  >>> wcs1.wcs.ctype = 'WAVE', 'HPLT-TAN', 'HPLN-TAN'
+  >>> wcs1.wcs.cunit = 'Angstrom', 'deg', 'deg'
+  >>> wcs1.wcs.cdelt = 0.2, 0.5, 0.4
+  >>> wcs1.wcs.crpix = 0, 2, 2
+  >>> wcs1.wcs.crval = 10, 0.5, 1
+
+  >>> wcs2 = astropy.wcs.WCS(naxis=3)
+  >>> wcs2.wcs.ctype = 'WAVE', 'HPLT-TAN', 'HPLN-TAN'
+  >>> wcs2.wcs.cunit = 'Angstrom', 'deg', 'deg'
+  >>> wcs2.wcs.cdelt = 0.2, 0.5, 0.4
+  >>> wcs2.wcs.crpix = 0, 2, 2
+  >>> wcs2.wcs.crval = 10, 0.5, 1
+
+  >>> # Instantiate NDCubes with supporting data.
+  >>> cube0 = NDCube(data0, wcs=wcs0)
+  >>> cube1 = NDCube(data1, wcs=wcs1)
+  >>> cube2 = NDCube(data2, wcs=wcs2)
```

Comment:
These three WCS objects are still essentially the same coordinates. Should we initialize the cubes with the same object or introduce some changes in the WCS?

I think passing on the same WCS instance will not highlight that `combine_cubes()` reprojects the cubes.

Reply:
Yeah I think we should probably change them. Maybe change the crpix values around a bit? (Also the HPLN/LT coordinates should really have CRVAL be 0,0).

Reply:
Any updates on this @adwaitbhope?

Reply:
Added a commit now